src/Rules/FunctionCallParametersCheck.php

```diff
@@ -109,20 +119,25 @@ public function check(
 					->build();
 			}
 			if ($hasUnpackedArgument && !$arg->unpack) {
-				$errors[] = RuleErrorBuilder::message('Unpacked argument (...) cannot be followed by a non-unpacked argument.')
-					->identifier('argument.nonUnpackAfterUnpacked')
-					->line($arg->getStartLine())
-					->nonIgnorable()
-					->build();
+				if ($argumentName !== null && $scope->getPhpVersion()->supportsNamedArgumentAfterUnpackedArgument()->yes()) {
+					if (in_array($argumentName, array_slice($functionParameterNames, 0, count($arguments)), true)) {
```

Comment:
What's going on here? Why the slice?

Reply:
`$functionParameterNames` is new and holds all parameter names the function accepts. `$arguments` is what this loop here is and was already building and includes all arguments that were passed via e.g. unpack or directly and so on. The slice based on the count of `$arguments` gives me the names of the parameters that were already "handled". So, if the current `$argumentName`, which is set in the case of handling a named argument for example, is inside there, it was already passed and passing it again would lead to a fatal error.

I had to think hard about this too now, thought about this already too long ago.. :D

Reply:
> I had to think hard about this too now

Exactly, this could use some refactoring. I don't get how "all handled parameters" tells us if one of them was unpacked or not. Also, we already report if an argument already overwrites an existing argument so instead we should make sure that place handles this instead of introducing new code to handle that.

See the `argument.duplicate` identifier.

Reply:
yeah, that was my initial thinking, but that stuff is somewhere different and handled later. I can make use of this, but need to pass information along this arguments array most likely. it's a bit complicated and ugly and I thought my addition has the least friction.

Reply:
but anyway, I can just try it out I guess, let's see. I most likely need to pass the information if this was a named argument and unpack already happened before it or so. then we should be good. blows up the array a bit with this edge case, but definitely more readable I think!

Reply:
ok, so I did something, WDYT? the tricky part is that the above handled `argument.nonUnpackAfterUnpacked` should not be created for such cases, while the new `argument.namedOverwriteAfterUnpacked` should and it is handled now below next to `argument.duplicate` which is then not necessary.